fontforgeexe/cvpalettes.c

```diff
@@ -1641,22 +1684,24 @@ return;
     base.trans = -1;
     GDrawSetFont(pixmap,layer2.font);
 
-    for ( i=(event->u.expose.rect.y-CV_LAYERS2_HEADER_HEIGHT)/CV_LAYERS2_LINE_HEIGHT;
-	    i<(event->u.expose.rect.y+event->u.expose.rect.height+CV_LAYERS2_LINE_HEIGHT-1-CV_LAYERS2_HEADER_HEIGHT)/CV_LAYERS2_LINE_HEIGHT;
-	    ++i ) {
+    // +2 for the defaults, +1 to show one extra (could be half showing it)
+    layerCount = layer2.visible_layers + 2 + 1;
+    layerCount = (layerCount > layer2.current_layers) ? layer2.current_layers : layerCount;
```

Comment:
I'll change it to a normal if statement. That said, I really dislike one liner if statements